### PR TITLE
Client heartbeat monitor should only apply for owner connections

### PR DESCRIPTION
**Pull request by vbekiaris**
Reasoning: the existing condition in client heartbeat monitor was
already checking whether this memberUUID is owner of the given
clientUUID, assuming that there exists a single connection between
a member and a client. However sometimes multiple connections may
exist leading the client heartbeat monitor to close a non-owner connection.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClientHeartbeatMonitor.java [hunk 1]
```diff
@@ -76,7 +76,9 @@ public void run() {
     }
 
     private void monitor(String memberUuid, ClientEndpointImpl clientEndpoint) {
-        if (clientEndpoint.isFirstConnection() && ClientType.CPP.equals(clientEndpoint.getClientType())) {
+        // C++ client sends heartbeat over its non-owner connections
+        // For other client types, disregard non-owner connections for heartbeat monitoring purposes
+        if (clientEndpoint.isOwnerConnection() == ClientType.CPP.equals(clientEndpoint.getClientType())) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClientHeartbeatMonitor.java]: Do not enforce heartbeat monitor:
- on owner connection of C++ client
- on non-owner connections of other clients

**Comment by mmedenjak** [comment]
@sancar @ihsandemir can we merge this PR? Anything missing?

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Pull request merged**